pkg/generators/markers.go

```diff
@@ -295,8 +325,9 @@ func putNestedValue(m map[string]any, k []string, v any) (map[string]any, error)
 	rest := k[1:]
 
 	if idxIdx := strings.Index(key, "["); idxIdx > -1 {
+		subscript := strings.Split(key[idxIdx+1:], "]")[0]
```

Comment:
Just want to double check, at the moment the array is only used for CEL right? Do we have other cases of array usage?

Reply:
correct this isnt used for anything else